src/sentry/ingest/ingest_consumer.py

```diff
@@ -73,9 +73,19 @@ def flush_batch(self, batch):
                 projects_to_fetch.add(message["project_id"])
 
                 if message_type == "event":
-                    other_messages.append((process_event, message))
-                elif message_type == "transaction":
-                    transactions.append(message)
+                    with metrics.timer("ingest_consumer.decode_event_json"):
+                        # XXX: Do not use CanonicalKeyDict here. This may break preprocess_event
+                        # which assumes that data passed in is a raw dictionary.
+                        message["payload"] = json.loads(message["payload"])
```

Comment:
Could we perhaps do this decoding _only_ when `options.get("store.transactions-celery")` is False?

Reply:
the decoding always has to happen for both message types, be it here or later in `process_event` or `process_transactions_batch`.